jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/profiles/ProfileCredentialProviderFactory.kt

```diff
@@ -306,6 +312,11 @@ class ProfileCredentialProviderFactory(private val ssoCache: SsoCache = diskCach
 
     companion object {
         private val LOG = getLogger<ProfileCredentialProviderFactory>()
+
+        val NEW_PROFILE_ADDED: Topic<ChangeConnectionSettingIfValid> = Topic.create(
+            "Change to newly added profile",
+            ChangeConnectionSettingIfValid::class.java
+        )
```

Comment:
take a look at `AwsConnectionManager` and `ToolkitCredentialsChangeListener`

Reply:
This notification topic is triggered only once when a single profile is added  and exists in the credentials file. The ToolkitCredentialsChangeListener triggers every time a provider is added.